assertj-core/src/test/java/org/assertj/core/internal/ConfigurableRecursiveFieldByFieldComparator_Test.java

```diff
@@ -87,4 +87,48 @@ void should_throw_an_NPE_if_given_RecursiveComparisonConfiguration_is_null() {
                    .hasMessage("RecursiveComparisonConfiguration must not be null");
   }
 
+  @Test
+  void should_success_ignore_non_existent_fields_when_configured() {
```

Comment:
I'm not completely sure what we are testing here, we check that the config has the expected `ignoreNonExistentFields` value but I don't get the point of checking `int compare = configurableRecursiveFieldByFieldComparator.compare(actual, other);`.

Instead of these two tests, could we add a test in `RecursiveComparisonAssert_isEqualTo_comparingOnlyFields_Test` and reproduce the test example in the initial issue https://github.com/assertj/assertj/issues/3806#issuecomment-2745230319 ? if the implementation is correct, the test should pass

Reply:
We could in addition also add a test with the code example from the javadoc (with BaseClass, Subclass1 and 2)

Thanks !